### DISCUSSION TITLE
Screening Method Sections

### DISCUSSION THREAD

**Discussion by SHogenboom** [asreview/asreview#1384] [Q&A]
Hi all,

I’m looking for people that may have already solved an issue I’m currently encountering. If not yet solved, any pointers in the right direction are of course also appreciated! The issue is as follows:

**I want to conduct a scoping/systematic review about the parameters of a psychological measurement.** Rather than wanting to know about outcome variables I’m thus interested how the measurement was used and which design choices were made. For example, how many stimuli were presented? Did participants provide verbal-/written answers? Etc. Because it is a very general measurement, it is **not** always mentioned in abstracts. I therefore want to screen based on the text from method sections. In theory this should not be a problem: the ‘abstract’ field can be filled with any body of text.

However… **how do I get to the method sections?** A manual copy-and-paste approach is not feasible as there are thousands of records to screen. I’m thus looking for an automated approach to extracting full text and/or method sections from DOIs. About two years ago I tried the `fulltext` package in R with mixed succes, but has since unfortunately been archived. Have not searched for Python solutions yet as I know little to no Python (though willing to learn).

Hoping for any pointers in the right direction,
Thanks in advance!

**Comment by Rensvandeschoot** [comment]
You are correct that ASReview allows you to add any type of text to the 'abstract' field, as explained in the data format documentation: https://asreview.readthedocs.io/en/latest/data_format.html. To achieve your goal, you can simply replace the text in the abstract column with the text of the methods section.

The main challenge, as you mentioned, is obtaining the full texts of the scientific papers and extracting the methods sections. To access full-text articles, you can use Unpaywall (https://unpaywall.org/), a free and open-source database of millions of open-access scholarly articles. By providing a DOI, you can obtain the full text of a paper if it's available as an open-access resource. Unpaywall offers an API that you can use to programmatically access the full text of articles: https://unpaywall.org/products/api.

Once you have the full text, you can build a scraper (e.g., using Python) to isolate the methods section. Keep in mind that the structure of scientific papers may vary between publishers and formats, so you might need to fine-tune your scraper to accurately extract the methods sections from various sources.

After obtaining the methods sections, you can compile them into a CSV file with the appropriate format (replacing the abstract column with the methods section text) and import it into ASReview.

I hope this helps you in your scoping/systematic review project!

**Reply by SHogenboom** [reply]
Thanks! Was not aware of Unpaywall so will definitely look into that.

Other recommendations I’ve (since) received:
* Start by searching databases which offer API utilities (e.g., Elsevier)
* Use EndNote (payed) find full text feature (will result in PDFs)
* Crossref also has an API which communicates with other APIs for full text extraction. 

Next week is for making any of these ideas work - so will post an update (with hopefully a solution)

**Reply by FelixWdm** [reply]
I asked around as well and got this response: "Here is an example for how you extract text from a pdf in R: https://bookdown.org/Maxine/tidy-text-mining/extracting-text-from-pdf-and-other-files.html - it uses the pdftools package (https://docs.ropensci.org/pdftools/). You could potentially build a corpus/dataset of sorts and then do some text-mining (https://www.tidytextmining.com/index.html)"

**Comment by rohitgarud** [comment]
Hi @SHogenboom, are you still looking for help in extracting particular information from the fulltext. I can help you with that if you can provide the pdfs and what information you want to extract. I am planning to develop an extension for the same, but for that, I will need an understanding of real-world scenarios and data if you can provide it. Thank you.

**Reply by SHogenboom** [reply]
Hi @rohitgarud; I am still looking for a solution yes, so any help would be appreciated! Have tried a range of things - but failed so far. Some examples of failure points I encountered:

[ Castell et al. (2018) ](https://sci-hub.st/10.1002/col.22168) 
Traditionally, articles go ‘Abstract’ > ‘Introduction’ > ‘Method’ > ‘Results’ > ‘Discussion’ > ‘Conclusion’. I’m trying to extract the method and results section based on pattern recognition. So my thought was to extract everything between the first mention of ‘method’ and ‘discussion. However, in this article this does not work because the ‘method’ section in forward references from within the introduction.

[Maier et al. (2008)](https://sci-hub.st/10.1177/0146167208323104)
This is an interesting usecase because it features multiple experiments. So far, general pattern recognition has been able to deal with scenarios like this with mixed results, because not all articles have a discussion per experiment.

Please let me know if these usecases are useful or wether you have more questions!

**Reply by rohitgarud** [reply]
Let's connect over mail.. can you please share say 10 papers having all the different scenarios including the above 2 failure cases and what you are trying to extract from these papers.. you can mail these things to rohit.garuda1992@gmail.com.. we will discuss further there..
